### PR TITLE
fix golint errors/warnings

### PR DESCRIPTION
**Pull request by brahmaroutu**
Addresses #14756
Signed-off-by: Srini Brahmaroutu srbrahma@us.ibm.com

### CODE REVIEW

File: api/server/server.go [hunk 1]
```diff
@@ -39,7 +39,8 @@ import (
 	"github.com/docker/docker/utils"
 )
 
-type ServerConfig struct {
+//APIServerConfig - provides the configuration for the API server
```
Comment (vdemeester) [inline api/server/server.go]: @brahmaroutu you don't need to add the `` - ``. I think the "look" of godoc will be a little weird.

Comment (runcom) [inline api/server/server.go]: agree with @vdemeester

File: api/server/server.go [hunk 2]
```diff
@@ -40,7 +40,8 @@ import (
 	"github.com/docker/docker/utils"
 )
 
-type ServerConfig struct {
+// APIServerConfig provides the configuration for the API server
+type APIServerConfig struct {
```
Comment (vdemeester) [inline api/server/server.go]: I'm not sure about this change, as ``APIServerConfig`` repeat the ``api/server`` part. ``Config`` would be simpler :blush:. Below you end up with ``apiserver.APIServerConfig`` in ``daemon.go`` which would look way cleaner like ``apiserver.Config``.

File: api/server/server.go [hunk 3]
```diff
@@ -49,15 +50,17 @@ type ServerConfig struct {
...
 	daemon  *daemon.Daemon
-	cfg     *ServerConfig
+	cfg     *APIServerConfig
 	router  *mux.Router
 	start   chan struct{}
 	servers []serverCloser
 }
 
-func New(cfg *ServerConfig) *Server {
+// New returns new instance of the server
```
Comment (vdemeester) [inline api/server/server.go]: "New returns a new instance of the server based on the specified configuration".

File: api/server/server.go [hunk 4]
```diff
@@ -117,19 +121,24 @@ func (s *Server) ServeApi(protoAddrs []string) error {
 	return nil
 }
 
-type HttpServer struct {
+// HTTPServer instance of http server and the listener
```
Comment (vdemeester) [inline api/server/server.go]: A verb is missing here, like "HTTPServer contains …"

File: api/server/server.go [hunk 5]
```diff
@@ -117,19 +121,24 @@ func (s *Server) ServeApi(protoAddrs []string) error {
...
+type HTTPServer struct {
 	srv *http.Server
 	l   net.Listener
 }
 
-func (s *HttpServer) Serve() error {
+// Serve start listening for inbound requests
```
Comment (vdemeester) [inline api/server/server.go]: s/start/starts/

File: api/server/server.go [hunk 6]
```diff
@@ -117,19 +121,24 @@ func (s *Server) ServeApi(protoAddrs []string) error {
...
+func (s *HTTPServer) Serve() error {
 	return s.srv.Serve(s.l)
 }
-func (s *HttpServer) Close() error {
+
+// Close close the HTTPServer
```
Comment (vdemeester) [inline api/server/server.go]: s/close/closes/

File: api/server/server.go [hunk 7]
```diff
@@ -117,19 +121,24 @@ func (s *Server) ServeApi(protoAddrs []string) error {
...
+func (s *HTTPServer) Close() error {
 	return s.l.Close()
 }
 
-type HttpApiFunc func(version version.Version, w http.ResponseWriter, r *http.Request, vars map[string]string) error
+// HTTPAPIFunc type definition of a api end point
```
Comment (vdemeester) [inline api/server/server.go]: Something like :

```go
// HTTPAPIFunc type is an adapter to allow the use of ordinary functions as Docker API endpoints.  
// Any function that has the appropriate signature can be register as a API endpoint (e.g. getVersion).
```

/cc @runcom @LK4D4

File: api/server/server.go [hunk 8]
```diff
@@ -154,7 +163,7 @@ func closeStreams(streams ...interface{}) {
 }
 
 // Check to make sure request's Content-Type is application/json
```
Comment (vdemeester) [inline api/server/server.go]: s/to make/makes/

File: docker/daemon.go [hunk 9]
```diff
@@ -115,7 +115,7 @@ func mainDaemon() {
 		}()
 	}
 
-	serverConfig := &apiserver.ServerConfig{
+	serverConfig := &apiserver.APIServerConfig{
```
Comment (vdemeester) [inline docker/daemon.go]: See above, I think ``apiserver.Config`` would look cleaner.

File: docker/daemon_unix.go [hunk 10]
```diff
@@ -15,7 +15,7 @@ import (
 	_ "github.com/docker/docker/daemon/execdriver/native"
 )
 
-func setPlatformServerConfig(serverConfig *apiserver.ServerConfig, daemonCfg *daemon.Config) *apiserver.ServerConfig {
+func setPlatformServerConfig(serverConfig *apiserver.APIServerConfig, daemonCfg *daemon.Config) *apiserver.APIServerConfig {
```
Comment (vdemeester) [inline docker/daemon_unix.go]: Same here, see above, I think ``apiserver.Config`` would look cleaner.

**Comment by vdemeester** [comment]
Few comments. See also https://github.com/docker/docker/pull/14751#issuecomment-123393428, you need a period at the end of the sentences for the func comments. :wink:

**Comment by brahmaroutu** [comment]
@vdemeester Please review.

File: api/server/server_linux.go [hunk 11]
```diff
@@ -67,6 +67,7 @@ func (s *Server) newServer(proto, addr string) ([]serverCloser, error) {
 	return res, nil
 }
 
+// AcceptConnections allow clients to connect to the API server.
```
Comment (vdemeester) [inline api/server/server_linux.go]: s/allow/allows/

**Comment by vdemeester** [comment]
@brahmaroutu Could you add the package ``api/server`` to ``./hack/make/validate-lint``.

File: api/server/server.go [hunk 12]
```diff
@@ -117,19 +121,25 @@ func (s *Server) ServeApi(protoAddrs []string) error {
 	return nil
 }
 
-type HttpServer struct {
+// HTTPServer contains an instance of http server and the listener.
```
Comment (LK4D4) [inline api/server/server.go]: let's write `http.Server` and `net.Listener` to have reference to real types in godoc.

File: api/server/server.go [hunk 13]
```diff
@@ -153,8 +163,8 @@ func closeStreams(streams ...interface{}) {
 	}
 }
 
-// Check to make sure request's Content-Type is application/json
-func checkForJson(r *http.Request) error {
+// Check makes sure that the request's Content-Type is application/json.
```
Comment (LK4D4) [inline api/server/server.go]: Maybe change for more `godoc`-ish `checkForJSON...` even if it's private.

File: api/server/server_linux.go [hunk 14]
```diff
@@ -67,6 +67,7 @@ func (s *Server) newServer(proto, addr string) ([]serverCloser, error) {
 	return res, nil
 }
 
+// AcceptConnections allows clients to connect to the API server.
```
Comment (LK4D4) [inline api/server/server_linux.go]: Here is crucial info, that we're telling daemon that we're ready to accept connections. That's why we need `d` parameter.

File: docker/daemon_unix.go [hunk 15]
```diff
@@ -15,7 +15,7 @@ import (
 	_ "github.com/docker/docker/daemon/execdriver/native"
 )
 
-func setPlatformServerConfig(serverConfig *apiserver.ServerConfig, daemonCfg *daemon.Config) *apiserver.ServerConfig {
+func setPlatformServerConfig(serverConfig *apiserver.Config, daemonCfg *daemon.Config) *apiserver.Config {
```
Comment (RichardScothern) [inline docker/daemon_unix.go]: Change this in daemon_windows.go too.

**Comment by RichardScothern** [comment]
One nit, but LGTM

File: docker/daemon_unix.go (hunk 15 above)
Comment (brahmaroutu) [inline docker/daemon_unix.go]: Wow, that is a good catch. Thanks @RichardScothern

**Comment by gdevillele** [comment]
LGTM

**Comment by tiborvass** [comment]
LGTM

**Pull request merged**